apps/site/src/app/(main)/guest-login/GuestLogin.tsx

```diff
@@ -3,9 +3,22 @@ import { redirect } from "next/navigation";
 import GuestLoginVerificationForm from "./components/GuestLoginVerificationForm";
 import getUserIdentity from "@/lib/utils/getUserIdentity";
 
-async function GuestLogin() {
+async function GuestLogin({
+	searchParams,
+}: {
+	searchParams?: {
+		return_to?: string;
+	};
+}) {
 	const identity = await getUserIdentity();
-	if (identity.uid !== null) redirect("/portal");
+	if (identity.uid !== null) {
+		if (searchParams?.return_to === undefined) {
+			redirect("/portal");
+		} else if (searchParams.return_to === "/portal") {
+			redirect(searchParams.return_to);
+		}
+		redirect(`/portal?return_to=${searchParams.return_to}`);
+	}
```

Comment:
Oh actually, reading #171 again, this is possible if the user navigates back in the browser. In that case, this can be redirect to the value or portal by default
```tsx
redirect(searchParams?.return_to ?? "/portal")
```